torch/csrc/jit/autodiff.cpp

```diff
@@ -37,14 +37,49 @@ bool isDifferentiable(Node * n) {
 
 static std::vector<Value*> gradientForNode(Node* node, ArrayRef<Value*> grad_values) {
   const auto build_sym_grad = [node](const std::vector<SymbolicVariable>& grads) -> std::vector<SymbolicVariable> {
-    auto inputs = node->inputs();
+    auto inputs = fmap<SymbolicVariable>(node->inputs());
+    auto outputs = fmap<SymbolicVariable>(node->outputs());
     switch(node->kind()) {
       case kadd:
         return {grads[0], grads[0]};
       case ksub:
         return {grads[0], -grads[0]};
       case kmul:
         return {grads[0] * inputs[1], grads[0] * inputs[0]};
+      case ksigmoid:
```

Comment:
Can you switch to the `grads.at(0)` variant? I ran into a hard-to-find bug because sometimes `kadd` and `kmul` only have one input (see variants that exist in aten_dispatch.cpp) I fixed this in grad executor for the existing gradient formulas but we should fix it for the new ones as well.